uboot-tools compile failed without openssl

### Describe the bug

After recent update, uboot-envtools is moved to uboot-tools and I recieve errors when compiling with default configuration.
```
make[4]: Entering directory '/home/user/openwrt/build_dir/target-arm_xscale_musl_eabi/u-boot-2025.01'
  HOSTLD  tools/dumpimage
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: tools/generated/lib/rsa/rsa-sign.o: in function `rsa_engine_get_priv_key':
rsa-sign.c:(.text+0x80): undefined reference to `ENGINE_get_id'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0x168): undefined reference to `ENGINE_load_private_key'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: tools/generated/lib/rsa/rsa-sign.o: in function `rsa_engine_remove':
rsa-sign.c:(.text+0x390): undefined reference to `ENGINE_finish'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0x39c): undefined reference to `ENGINE_free'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: tools/generated/lib/rsa/rsa-sign.o: in function `rsa_engine_init.constprop.0':
rsa-sign.c:(.text+0x3ac): undefined reference to `ENGINE_load_builtin_engines'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0x3b4): undefined reference to `ENGINE_by_id'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0x3e8): undefined reference to `ENGINE_init'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0x408): undefined reference to `ENGINE_free'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0x414): undefined reference to `ENGINE_set_default_RSA'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0x434): undefined reference to `ENGINE_finish'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0x458): undefined reference to `ENGINE_ctrl_cmd_string'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: tools/generated/lib/rsa/rsa-sign.o: in function `rsa_add_verify_data':
rsa-sign.c:(.text+0xb48): undefined reference to `ENGINE_get_id'
/home/user/openwrt/staging_dir/toolchain-arm_xscale_gcc-13.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: rsa-sign.c:(.text+0xbf8): undefined reference to `ENGINE_load_public_key'
collect2: error: ld returned 1 exit status
make[5]: *** [scripts/Makefile.host:104: tools/dumpimage] Error 1
make[4]: *** [Makefile:1915: tools] Error 2
make[4]: Leaving directory '/home/user/openwrt/build_dir/target-arm_xscale_musl_eabi/u-boot-2025.01'
make[3]: *** [Makefile:121: /home/user/openwrt/build_dir/target-arm_xscale_musl_eabi/u-boot-2025.01/.built] Error 2
```
This happens when only mbedtls is selected.
If I select openssl too, it will gone.

@TheRootED24 


### OpenWrt version

Snapshot

### OpenWrt release

Snapshot

### OpenWrt target/subtarget

kirkwood

### Device

all devices

### Image kind

Self-built image

### Steps to reproduce

_No response_

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

_No response_

### Diffconfig

```text

```

### Terms

- [x] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/package/boot/uboot-tools/Makefile
+++ b/package/boot/uboot-tools/Makefile
@@ -59,8 +59,8 @@ define Package/uboot-envtools/conffiles
 endef
 
 define Build/Configure
-	$(MAKE) -C $(PKG_BUILD_DIR) tools-only_defconfig
-	$(MAKE) -C $(PKG_BUILD_DIR) syncconfig
+	$(call Build/Compile/Default,tools-only_defconfig)
+	$(call Build/Compile/Default,syncconfig)
 	$(SED) 's/CONFIG_TOOLS_LIBCRYPTO=y/# CONFIG_TOOLS_LIBCRYPTO is not set/' $(PKG_BUILD_DIR)/.config
 endef
 
@@ -71,8 +71,14 @@ MAKE_FLAGS += \
 	NO_PYTHON=1
 
 define Build/Compile
+
+ifneq ($(CONFIG_PACKAGE_uboot-envtools),)
 	$(call Build/Compile/Default,envtools)
+endif
+ifneq ($(CONFIG_PACKAGE_dumpimage),)
 	$(call Build/Compile/Default,cross_tools)
+endif
+
 endef
 
 define Package/dumpimage/install

--- a/package/boot/uboot-tools/Makefile
+++ b/package/boot/uboot-tools/Makefile
@@ -22,19 +22,6 @@ PKG_BUILD_PARALLEL:=1
 
 include $(INCLUDE_DIR)/package.mk
 
-define Package/uboot-tools
-	SECTION:=utils
-	CATEGORY:=Utilities
-	SUBMENU:=Boot Loaders
-	TITLE:=U-Boot bootloader Tools
-	URL:=http://www.denx.de/wiki/U-Boot
-endef
-
-define Package/uboot-tools/description
-	U-Boot tools are a collection of utilities designed
-	to work with the U-Boot bootloader,
-endef
-
 define Package/dumpimage
 	SECTION:=utils
 	CATEGORY:=Utilities

